source/Cosmos.System2/Network/IPv4/TCP/TCPClient.cs

```diff
@@ -15,78 +15,19 @@
 namespace Cosmos.System.Network.IPv4.TCP
 {
     /// <summary>
-    /// TCP Connection status
-    /// </summary>
-    public enum Status
-    {
-        LISTEN,
-        SYN_SENT,
-        SYN_RECEIVED,
-        ESTABLISHED,
-        FIN_WAIT1,
-        FIN_WAIT2,
-        CLOSE_WAIT,
-        CLOSING,
-        LAST_ACK,
-        TIME_WAIT,
-        CLOSED
-    }
-
-    /// <summary>
-    /// TCPClient class. Used to manage the TCP connection to a client.
+    /// TcpClient class. Used to manage the TCP connection to a server.
     /// </summary>
     public class TcpClient : IDisposable
     {
         /// <summary>
-        /// Clients dictionary.
-        /// </summary>
-        private static Dictionary<uint, TcpClient> clients;
-
-        /// <summary>
-        /// Local port.
-        /// </summary>
-        private int localPort;
-        /// <summary>
-        /// Source address.
-        /// </summary>
-        internal Address source;
-        /// <summary>
-        /// Destination address.
+        /// Tcp State machine.
         /// </summary>
-        internal Address destination;
-        /// <summary>
-        /// Destination port.
-        /// </summary>
-        private int destinationPort;
+        internal Tcp StateMachine;
 
         /// <summary>
-        /// RX buffer queue.
-        /// </summary>
-        internal Queue<TCPPacket> rxBuffer;
-
-        /// <summary>
-        /// Connection status.
-        /// </summary>
-        internal Status Status;
-
-        /// <summary>
-        /// Connection Acknowledgement number.
-        /// </summary>
-        internal uint AckNumber;
-
-        /// <summary>
-        /// Connection Sequence number.
-        /// </summary>
-        internal uint SequenceNumber;
-
-        /// <summary>
-        /// Last recveived Connection Sequence number.
+        /// Clients dictionary.
         /// </summary>
-        private uint LastSequenceNumber;
-
-        private bool WaitingAck = false;
-
-        private byte[] data;
+        private static Dictionary<uint, TcpClient> clients;
```

Comment:
Do we need the list of clients here? Wouldn't it be better if every Listener keeps a list of clients which connected via it?
Looking at the code, I don't really see the need for the Dictionary here.

Reply:
It's a static list containing all tcp clients linked by their ports, it's necessary to handle multiple connections

Reply:
From where is it used?

Reply:
When a new packet is received here: https://github.com/CosmosOS/Cosmos/blob/feature/tcp/source/Cosmos.System2/Network/IPv4/TCP/TCPPacket.cs#L57